File: WeLegends/andiag-mvp/src/main/java/es/coru/andiag/andiag_mvp/base/BasePresenter.java
package es.coru.andiag.andiag_mvp.base;

import android.content.Context;

import es.coru.andiag.andiag_mvp.interfaces.BaseView;

/**
 * Created by Canalejas on 10/12/2016.
 */

public abstract class BasePresenter<V extends Context & BaseView> {
    private final static String TAG = BasePresenter.class.getSimpleName();

    private V mView;

    public V getView() {
        return mView;
    }

    public final void attach(V view) {
        this.mView = view;
        onViewAttached();
    }

    public final void detach() {
        this.mView = null;
        onViewDetached();
    }

    public abstract void onViewAttached();

    public abstract void onViewDetached();

}

File: WeLegends/app/src/main/java/es/coru/andiag/welegends/common/base/BasePresenter.kt
package es.coru.andiag.welegends.common.base

/**
 * Created by Canalejas on 08/12/2016.
 */


abstract class BasePresenter<V> {

    private var view: V? = null

    fun attach(view: V) {
        this.view = view
        onAttach()
    }

    open fun onAttach() {
    }

    fun detach() {
        onDetach()
        view = null
    }

    open fun onDetach() {
    }

    protected fun getView() = view!!

}
